### PR TITLE
rgw/rgw_rados: set pg_autoscale_bias=4 for omap pools

### PR DESCRIPTION
**Pull request by liewegas**
Not sure if this is the best approach here....

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -174,6 +176,18 @@ int rgw_init_ioctx(librados::Rados *rados, const rgw_pool& pool, IoCtx& ioctx, b
     if (r < 0 && r != -EEXIST) {
       return r;
     }
+    if (mostly_omap) {
+      // set pg_autoscale_bias
+      bufferlist inbl;
+      int r = rados->mon_command(
+	"{\"prefix\": \"osd pool set\", \"pool\": \"" +
+	pool.name + "\", \"var\": \"pg_autoscale_bias\": \"4.0\"}",
```
Comment (mattbenjamin) [inline src/rgw/rgw_rados.cc]: maybe we could lift the 4.0 value out to a constant (not a #define)?

Comment (liewegas) [reply src/rgw/rgw_rados.cc]: updated!  added options for both pg_num_min and autoscale_bias

**Comment by cbodley** [comment]
thanks @liewegas - i see that rgw_init_ioctx() is currently duplicated in services/svc_rados.cc as init_ioctx() though, so there are some code paths that this won't cover

**Comment by liewegas** [comment]
> thanks @liewegas - i see that rgw_init_ioctx() is currently duplicated in services/svc_rados.cc as init_ioctx() though, so there are some code paths that this won't cover

Should i just duplicate this change there, or is there some cleanup/consolidation we should do first?

**Comment by cbodley** [comment]
hmm I think the goal is to keep any dependencies on rgw_rados.* out of the services, so i'd suggest moving this `rgw_init_ioctx()` to rgw_tools.h/cc

services/svc_rados.cc already includes rgw_tools.h, so we could remove its copy of `init_ioctx()`

**Comment by cbodley** [comment]
@liewegas i pushed a commit for that to https://github.com/ceph/ceph/pull/27410 - feel free to cherry-pick it here

File: src/rgw/rgw_tools.cc [hunk 2]
```diff
@@ -30,6 +32,69 @@
...
+	dout(10) << __func__ << " warning: failed to set pg_autoscale_bias on "
+		 << pool.name << dendl;
+      }
+      // set pg_num_min
+      int min = g_conf().get_val<uint64_t>("rgw_rados_pool_pg_num_min");
+      r = rados->mon_command(
+	"{\"prefix\": \"osd pool set\", \"pool\": \"" +
+	pool.name + "\", \"var\": \"pg_num_min\": \"" +
+	stringify(min) + "\"}",
+	inbl, NULL, NULL);
```
Comment (cbodley) [inline src/rgw/rgw_tools.cc]: @liewegas i'm seeing some error messages from these commands on a vstart cluster:
> 2019-05-03T11:46:07.569-0400 7fde1b479700  1 -- 10.17.151.111:0/2932979858 <== mon.0 v2:10.17.151.111:6820/0 12 ==== mon_command_ack([{"prefix": "osd pool set", "pool": "a1.rgw.buckets.index", "var": "pg_autoscale_bias": "4"}]=-22 unparseable JSON {"prefix": "osd pool set", "pool": "a1.rgw.buckets.index", "var": "pg_autoscale_bias": "4"} v0) v1 ==== 233+0+0 (secure 0 0 0) 0x560c4796bb00 con 0x560c47cced00
2019-05-03T11:46:07.570-0400 7fde1b479700  1 -- 10.17.151.111:0/2932979858 <== mon.0 v2:10.17.151.111:6820/0 13 ==== mon_command_ack([{"prefix": "osd pool set", "pool": "a1.rgw.buckets.index", "var": "pg_num_min": "8"}]=-22 unparseable JSON {"prefix": "osd pool set", "pool": "a1.rgw.buckets.index", "var": "pg_num_min": "8"} v0) v1 ==== 219+0+0 (secure 0 0 0) 0x560c4796bd40 con 0x560c47cced00

Comment (liewegas) [reply src/rgw/rgw_tools.cc]: see https://github.com/ceph/ceph/pull/27967

**Pull request merged**